shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -84,6 +84,29 @@ @interface FlutterEngine () <FlutterBinaryMessenger>
 
 - (nullable FlutterViewController*)viewControllerForId:(uint64_t)viewId;
 
+/**
+ * An internal method that adds view controller with the given ID.
+ *
+ * This method assigns the controller with the ID, puts the controller into the
+ * map, and does assertions related to the default view ID.
+ */
+- (void)registerViewController:(FlutterViewController*)controller forId:(uint64_t)viewId;
```

Comment:
Is `withId:` more idiomatic? Same comment for `deregisterViewControllerForId` below

```suggestion
- (void)registerViewController:(FlutterViewController*)controller withId:(uint64_t)viewId;
```

Reply:
`For` is better here. In Obj-C, typically you'll see "withFoo" in constructors and class factory methods (`initWithView:` etc.). For getters, you'll see a lot of forFoo: (e.g. `getFirstRectForRange:`, `getObjectValue:forString:range:error:`).